lib/util/index.js

```diff
@@ -278,6 +278,15 @@ function delegateMethods(source, wrapper, target) {
   }
 }
 
+/**
+ * Whether the given argument is a non-array object.
+ * @param {*} object
+ * @return {boolean}
+ */
+function isNonArrayObject(object) {
```

Comment:
Most of the times, we would want to also exclude `null` value as an object. It would be useful if we folded null check inside this function as well. 

not sure what would you call the function. how about `isSimpleObject` 
```
function isSimpleObject(subject) {
   return subject !== null && typeof subject === 'object' && !Array.isArray(subject);
}
```

Reply:
@makarandp0 I have used this method in places where `null` is a valid value. So, for now, I think I'm more comfortable with the null check being outside the function.